test/test_rest_vol.c

```diff
@@ -2493,16 +2493,16 @@ test_create_attribute_on_root(void)
 #endif
 
         if ((/*attr_id = */ H5Aopen_by_idx(file_id, "/", H5_INDEX_NAME, H5_ITER_INC, 0, H5P_DEFAULT,
```

Comment:
The code above this point that surrounds calls to `H5Aclose` with `#if 0` should be re-enabled so that the IDs for the attributes from the previous tests get closed correctly. Then, all these lines where `attr_id` and `attr_id2` are commented out should be re-enabled so that the attribute IDs are properly assigned and later closed. Otherwise, we're leaking several IDs from the library. There are at least two instances of this in this file that I can see.